src/components/PromotionalSash/PromotionalSash.js

```diff
@@ -0,0 +1,61 @@
+import styled from '@emotion/styled';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { themeGet } from 'styled-system';
+import { withTheme } from 'emotion-theming';
+import { Box, Icon, Flex, Text } from '../';
+
+const Badge = styled(Box)`
+  text-align: center;
+  display: inline-block;
+  text-transform: uppercase;
+  white-space: nowrap;
+  letter-spacing: 1px;
+  overflow: hidden;
+  color: ${themeGet('colors.ui.badgeText')};
+  background-color: ${themeGet('colors.ui.badgeBackground')};
+  padding: 0 ${themeGet('space.1')};
+`;
+
+const BadgeText = styled(Text)`
+  font-weight: ${themeGet('fontWeights.bold')};
+  overflow: hidden;
+  text-overflow: ellipsis;
+`;
+
+const PromotionalSash = withTheme(({ promotionName, fontSize, promoIcons }) => {
+  const iconName = promoIcons[promotionName.toLowerCase()];
+
+  return (
+    <Badge>
+      <Flex alignItems="center">
+        {iconName && (
+          <Icon name={iconName} size={16} color="greys.charcoal" mr={1} />
+        )}
+        <BadgeText fontSize={fontSize}>{promotionName}</BadgeText>
+      </Flex>
+    </Badge>
+  );
+});
+
+PromotionalSash.propTypes = {
+  promotionName: PropTypes.string.isRequired,
```

Comment:
There is a similar `<Sash />` component in holidays codebase, but we don't have `promotionName` hence no icon
```suggestion
  promotionName: PropTypes.string,
```

Reply:
Actually thinking should we put product related logic into roo ui?

Reply:
Thanks @KieraDOG! Fair point. I've taken out the `isRequired` and renamed `promotionName` to just `text` so that it's more generic.

One of the purposes of the `<PromotionalSash />` is to render an icon if a mapping has been provided. I'm happy to remove the `defaultProps` for `icons` and let the consumer provide their promotion-icon mapping to remove the product-related logic.

Reply:
Yes, I think it would be great to decouple the logic to get icon from this rendering component